app/models/manageiq/providers/amazon/agent_coordinator.rb

```diff
@@ -0,0 +1,423 @@
+        next
+      end
+    end
+
+    _log.error("Failed to activate agents: #{agent_ids}.")
+  end
+
+  def deploy_agent
+    _log.info("Deploying agent ...")
+    @deploying = true
+
+    kp = find_or_create_keypair
+    zone_name = ec2.client.describe_availability_zones.availability_zones[0].zone_name
+    subnets = get_subnets(zone_name)
+    raise "No subnet_id is available for #{zone_name}!" if subnets.empty?
+    security_group_id = find_or_create_security_group(subnets[0].vpc_id)
+    find_or_create_profile
+
+    instance                = ec2.create_instances(
+      :image_id             => get_agent_image_id,
+      :min_count            => 1,
+      :max_count            => 1,
+      :key_name             => kp.name,
+      :security_group_ids   => [security_group_id],
+      :instance_type        => 't2.micro',
+      :placement            => {
+        :availability_zone => zone_name
+      },
+      :subnet_id            => subnets[0].subnet_id,
+      :iam_instance_profile => {
+        :name => label
+      },
+      :tag_specifications   => [{
+        :resource_type => "instance",
+        :tags          => [{
+          :key   => "Name",
+          :value => label
+        }]
+      }]
+    )
+    ec2.client.wait_until(:instance_status_ok, :instance_ids => [instance[0].id])
+
+    _log.info("Start to load SSA application, this may take a while ...")
+
+    setup_agent(instance[0])
+    _log.info("SSA agent is ready to receive requests.")
+
+    instance[0].id
+  end
+
+  def setup_agent(instance)
+    # register if needed
+    register(instance)
+
+    # prepare work directory
+    ssh_commands(instance, ["sudo mkdir -p #{WORK_DIR}; sudo chmod go+w #{WORK_DIR}"], agent_ami_login_user)
+    # scp the default setting yaml file
+    create_config_yaml("config.yml")
+    scp_file("config.yml", WORK_DIR.to_s, instance, agent_ami_login_user)
+
+    # run docker image
+    command_line = "sudo docker run -d --restart=always -v /dev:/host_dev -v #{WORK_DIR}/config.yml:#{WORK_DIR}/config.yml --privileged #{agent_docker_name}"
+    ssh_commands(instance, [command_line.to_s], agent_ami_login_user)
+  end
+
+  # TODO: for downstream
+  def register(instance)
+  end
+
+  # Get Key Pair for SSH. Create a new one if not exists.
+  def find_or_create_keypair(keypair_name = label)
+    get_keypair(keypair_name) || begin
+      _log.info("KeyPair #{keypair_name} will be created!")
+      ManageIQ::Providers::CloudManager::AuthKeyPair.create_key_pair(@ems.id, :key_name => keypair_name)
+    end
+  end
+
+  def get_keypair(keypair_name = label)
+    kps = @ems.authentications.where(:name => keypair_name)
+    kps.any? ? kps[0] : nil
+  end
+
+  def find_or_create_profile(profile_name = label, role_name = label)
+    ssa_profile = iam.instance_profile(profile_name)
+    ssa_profile = iam.create_instance_profile(:instance_profile_name => profile_name) unless ssa_profile.exists?
```

Comment:
You can tag a `.presence` onto the line above then use `||=` here and drop the `unless ssa_profile.exists?`

Reply:
I'm assuming `iam.instance_profile(profile_name)` is returning a string, so the `.presence` may not be needed?